lib/apollo/hooks/actions/sendAnswerAndGetNextQuestion.ts

```diff
@@ -0,0 +1,52 @@
+import SendAnswerAndGetNextQuestion from 'graphql/mutations/sendAnswerAndGetNextQuestion.graphql';
+import { useMutation } from '@apollo/client';
+
+import { useNotifier } from 'contexts/NotifierContext';
+import { useCallback } from 'react';
+import Question from 'domain/Question';
+
+type SendAnswerAndGetNextQuestionProps = {
+  gameId: string | number;
+  value: string | undefined;
+};
+
+type SendAnswerAndGetNextQuestionData = {
+  sendAnswerAndGetNextQuestion: {
+    correctAnswerValue: string;
+    question: Question;
+    correctAnswersCount: number;
+  };
+};
+
+type SendAnswerAndGetNextQuestionInputVariable = SendAnswerAndGetNextQuestionProps;
+
+type SendAnswerAndGetNextQuestionVariables = {
+  input: SendAnswerAndGetNextQuestionInputVariable;
+};
+
+const useSendAnswerAndGetNextQuestion = () => {
+  const { setError } = useNotifier();
+
+  const [mutation, sendAnswerAndGetNextQuestionState] = useMutation<
+    SendAnswerAndGetNextQuestionData,
+    SendAnswerAndGetNextQuestionVariables
+  >(SendAnswerAndGetNextQuestion, {
+    onError: (error) => {
+      if (setError) setError(error);
+    },
+  });
+
+  const sendAnswerAndGetNextQuestion = useCallback(
+    ({ gameId, value }: SendAnswerAndGetNextQuestionProps) => {
+      const sendAnswerAndGetNextQuestionInput = { gameId: +gameId, value };
+      return mutation({
+        variables: { input: sendAnswerAndGetNextQuestionInput },
+      });
+    },
+    [mutation],
+  );
+
+  return [sendAnswerAndGetNextQuestion, sendAnswerAndGetNextQuestionState] as const;
```

Comment:
may be it should be better to prepare data from sendAnswerAndGetNextQuestionState and return only question from `sendAnswerAndGetNextQuestionState.data.sendAnswerAndGetNextQuestion.question`

Reply:
We also need `data.sendAnswerAndGetNextQuestion.correctAnswerValue` so i'd like to leave it like this and `data.sendAnswerAndGetNextQuestion.question`.